pkg/adaptor/cloud/azure/provider.go

```diff
@@ -450,3 +387,108 @@ func (p *azureProvider) updateInstanceSizeSpecList() error {
+				SecurityEncryptionType: to.Ptr(armcompute.SecurityEncryptionTypesVMGuestStateOnly),
+			},
+		}
+
+		securityProfile = &armcompute.SecurityProfile{
+			SecurityType: to.Ptr(armcompute.SecurityTypesConfidentialVM),
+			UefiSettings: &armcompute.UefiSettings{
+				SecureBootEnabled: to.Ptr(true),
+				VTpmEnabled:       to.Ptr(true),
+			},
+		}
+	} else {
+		managedDiskParams = &armcompute.ManagedDiskParameters{
+			StorageAccountType: to.Ptr(armcompute.StorageAccountTypesStandardLRS),
+		}
+
+		securityProfile = nil
+	}
+
+	imgRef := &armcompute.ImageReference{
+		ID: to.Ptr(p.serviceConfig.ImageId),
+	}
+	if strings.HasPrefix(p.serviceConfig.ImageId, "/CommunityGalleries/") {
+		imgRef = &armcompute.ImageReference{
+			CommunityGalleryImageID: to.Ptr(p.serviceConfig.ImageId),
+		}
+	}
+
+	// Add tags to the instance
+	tags := map[string]*string{}
+
+	// Add custom tags from serviceConfig.Tags to the instance
+	for k, v := range p.serviceConfig.Tags {
+		tags[k] = to.Ptr(v)
+	}
+
+	vmParameters := armcompute.VirtualMachine{
+		Location: to.Ptr(p.serviceConfig.Region),
+		Properties: &armcompute.VirtualMachineProperties{
+			HardwareProfile: &armcompute.HardwareProfile{
+				VMSize: to.Ptr(armcompute.VirtualMachineSizeTypes(instanceSize)),
+			},
+			StorageProfile: &armcompute.StorageProfile{
+				ImageReference: imgRef,
+				OSDisk: &armcompute.OSDisk{
+					Name:         to.Ptr(diskName),
+					CreateOption: to.Ptr(armcompute.DiskCreateOptionTypesFromImage),
+					Caching:      to.Ptr(armcompute.CachingTypesReadWrite),
+					DeleteOption: to.Ptr(armcompute.DiskDeleteOptionTypesDelete),
+					ManagedDisk:  managedDiskParams,
+				},
+			},
+			OSProfile: &armcompute.OSProfile{
+				AdminUsername: to.Ptr(p.serviceConfig.SSHUserName),
+				ComputerName:  to.Ptr(instanceName),
+				LinuxConfiguration: &armcompute.LinuxConfiguration{
+					DisablePasswordAuthentication: to.Ptr(true),
+					//TBD: replace with a suitable mechanism to use precreated SSH key
+					SSH: &armcompute.SSHConfiguration{
+						PublicKeys: []*armcompute.SSHPublicKey{{
+							Path:    to.Ptr(fmt.Sprintf("/home/%s/.ssh/authorized_keys", p.serviceConfig.SSHUserName)),
+							KeyData: to.Ptr(string(sshBytes)),
+						}},
+					},
+				},
+			},
+			NetworkProfile: &armcompute.NetworkProfile{
+				NetworkInterfaces: []*armcompute.NetworkInterfaceReference{
+					{
+						ID: vmNIC.ID,
+						Properties: &armcompute.NetworkInterfaceReferenceProperties{
+							DeleteOption: to.Ptr(armcompute.DeleteOptionsDelete),
+						},
+					},
+				},
+			},
+			SecurityProfile: securityProfile,
+		},
+		// Add tags to the instance
+		Tags: tags,
+	}
+
+	// If EnableUserData is set to false then set OSProfile.CustomData to b64EncData and
+	// armcompute.VirtualMachine.Properties.UserData to nil
+	// If EnableUserData is set to true then set armcompute.VirtualMachine.Properties.UserData to b64EncData and
+	// OSProfile.CustomData to nil
```

Comment:
I wouldn't differentiate between CustomData + UserData. As I understand it, CustomData is merely an older way to pass-in user data via a UDF filesystem to a VM. So, what we are actually doing is writing the file in Cloud-Config format for cloud-init or in our own specific format (as raw json for example). 

Whether user-data/custom-data is picked and parsed by cloud-init depends on the first line:

![image](https://github.com/confidential-containers/cloud-api-adaptor/assets/273280/520667a8-e8ef-48af-926f-5d5b6bf61729)

[see here](https://learn.microsoft.com/en-us/azure/virtual-machines/linux/using-cloud-init#deploying-a-cloud-init-enabled-virtual-machine)

Hence I would suggest to always in UserData